tls/s2n_server_new_session_ticket.c

```diff
@@ -82,8 +82,10 @@ int s2n_server_nst_send(struct s2n_connection *conn)
     uint32_t lifetime_hint_in_secs =
             (conn->config->encrypt_decrypt_key_lifetime_in_nanos + conn->config->decrypt_key_lifetime_in_nanos) / ONE_SEC_IN_NANOS;
 
-    /* When server changes it's mind mid handshake send lifetime hint and session ticket length as zero */
-    if (!conn->config->use_tickets) {
+    /* When server changes it's mind mid handshake, or if session key used to encrypt session ticket
+     * is expired, send lifetime hint and session ticket length as zero 
+     */
+    if (!conn->config->use_tickets || !s2n_config_is_encrypt_decrypt_key_available(conn->config)) {
```

Comment:
Your PR #4513 makes this comment irrelevant, but I'll mention it since it would be good to think about. s2n_config_is_encrypt_decrypt_key_available() returns 1 if a key exists, 0 if a key doesn't exist, and -1 if an error occurs. Therefore the correct usage of the function here would be `s2n_config_is_encrypt_decrypt_key_available() != 0`. The way its written here means that your check will fail if an error occurs, when we actually want it to succeed and send a zero-length ticket.